client-implementation.md

```diff
@@ -0,0 +1,67 @@
+# Client Implementations
+
+This document describes client implementation recommendations for interoperability with registry servers and other clients using the OCI Distribution Spec.
+
+## Backwards Compatibility
+
+Client implementations SHOULD support registries that implement partial or older versions of the OCI Distribution Spec.
+This section describes how client fallback procedures when an API is not available.
+
+### Referrers API
+
+The Referrers API here is described by [Listing Referrers](spec.md#listing-referrers) and [end-12a](spec.md#endpoints).
+
+A client that pushes an Image or Artifact manifest with a defined `Refers` field MUST verify the Referrers API is available.
+A client querying the Referrers API and receiving a 404 MUST fallback to using an Index pushed to a tag described by the following schema.
+
+#### Referrers Tag Schema
+
+```text
+<alg>-<ref>
+```
+
+- `<alg>`: the digest algorithm (e.g. `sha256` or `sha512`)
+- `<ref>`: the digest from the `refers` field (limit of 64 characters)
+
+For example, a manifest with the `Refers` field digest set to `sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa` in the `registry.example.org/project` repository would have a expect an Index at `registry.example.org/project:sha256-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
+
+#### Pushing Manifests with Refers
+
+When pushing an Image or Artifact manifest with the `Refers` field and the Referrers API returns a 404, the client MUST:
+
+1. Pull the Index using the tag schema.
```

Comment:
```suggestion
1. Pull the manifest using the digest from the Refers field.
```

Reply:
This isn't pulling the target manifest. It's pulling an Index of referrers to the target manifest.

Reply:
```suggestion
1. Pull the Image Index using the referrers tag schema.
```

Reply:
for what it's worth the image index is described as a higher level manifest.. In other portions of this spec we've tried to use the more generic manifest term if and when it did make sense. I see now that this probably doesn't make sense in this case :-)